crates/svm-runtime-c-api/src/byte_array.rs

```diff
@@ -56,31 +61,43 @@ impl Default for svm_byte_array {
 /// use std::{convert::TryFrom, string::FromUtf8Error};
 /// use svm_runtime_c_api::svm_byte_array;
 ///
-/// let s = "Hello World!";
-/// let ptr = s.as_ptr();
-/// let array: svm_byte_array = s.into();
-/// assert_eq!(ptr, array.bytes);
+/// let s1 = "Hello World!";
+/// let bytes: svm_byte_array = s1.into();
+/// assert_eq!(s1.as_ptr(), bytes.bytes);
+/// assert_eq!(s1.len() as u32, bytes.length);
+/// assert_eq!(s1.len() as u32, bytes.capacity);
 ///
-/// let s = String::try_from(array);
-/// assert_eq!("Hello World!".to_string(), s.unwrap());
+/// let s2 = String::try_from(bytes).unwrap();
+/// assert_eq!(s1.to_string(), s2);
 /// ```
 ///
 impl From<&str> for svm_byte_array {
```

Comment:
I believe we'll have now to manage without `From<&str>`
since a slice in Rust don't expose a `capacity` (it makes sense since it's an implementation detail)
the new code should have only `From<...>` from types we can extract it's `capacity`.

I'd suggest that you remove it and let's see where the compiler yells.

Reply:
Yes, I was thinking that it's unsafe to continue supporting it, but wasn't sure if we want to keep it anyway.

Compilation is fine after the removal, but some tests are failing. Will fix it.